src/main/scala/org/ergoplatform/nodeView/wallet/persistence/WalletRegistry.scala

```diff
@@ -273,22 +269,45 @@ class WalletRegistry(store: HybridLDBKVStore)(ws: WalletSettings) extends Scorex
   /**
     * Updates scans of a box stored in the wallet database,
     * puts the box into the database if it is not there
+    * removes the box from the database if its there and scanIds are empty
     *
-    * @param scanIds
+    * @param newScans
     * @param box
     * @return
     */
-  def updateScans(scanIds: Set[ScanId], box: ErgoBox): Try[Unit] = Try {
+  def updateScans(newScans: Set[ScanId], box: ErgoBox): Try[Unit] = Try {
     val bag0 = KeyValuePairsBag(toInsert = Seq.empty, toRemove = Seq.empty)
```

Comment:
Can we call it emptyBag and even move out of the method (if it is immutable) to save allocations and also clarify logic.

Reply:
Done, swapped with already existing KeyValuePairsBag.empty